### PR TITLE
Allow inviting users even if profile information cannot be retrieved

### LINKED ISSUE #3306: Fix user invitation in case of restricted profile api
If the server is configured as `limit_profile_requests_to_users_who_share_rooms: true` then profile API will not return user information, instead, the response will be `403`. Invitation of valid user ids should be allowed.

### PR DESCRIPTION
**Pull request by onurays**
Fixes #3306.

### CODE REVIEW

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/profile/ProfileService.kt [hunk 1]
```diff
@@ -66,6 +66,7 @@ interface ProfileService {
     /**
      * Get the combined profile information for this user.
      * This may return keys which are not limited to displayname or avatar_url.
+     * If server is configured as limit_profile_requests_to_users_who_share_rooms: true then response will be HTTP 403.
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/profile/ProfileService.kt:69]:
```suggestion
     * If server is configured as limit_profile_requests_to_users_who_share_rooms: true then response can be HTTP 403.
```

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/profile/ProfileAPI.kt [hunk 2]
```diff
@@ -33,6 +33,7 @@ internal interface ProfileAPI {
      * Get the combined profile information for this user.
      * This API may be used to fetch the user's own profile information or other users; either locally or on remote homeservers.
      * This API may return keys which are not limited to displayname or avatar_url.
+     * If server is configured as limit_profile_requests_to_users_who_share_rooms: true then response will be HTTP 403.
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/profile/ProfileAPI.kt:36]:
```suggestion
     * If server is configured as limit_profile_requests_to_users_who_share_rooms: true then response can be HTTP 403.
```

**Review by bmarty — APPROVED** [review]
Just some remark about the doc, else LGTM

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/profile/ProfileAPI.kt (hunk 2 above)
Comment (onurays) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/profile/ProfileAPI.kt]: Done.

File: vector/src/main/java/im/vector/app/features/userdirectory/UserListViewModel.kt [hunk 3]
```diff
@@ -137,7 +136,15 @@ class UserListViewModel @AssistedInject constructor(@Assisted initialState: User
                                                 avatarUrl = json[ProfileService.AVATAR_URL_KEY] as? String
                                         ).toOptional()
                                     }
-                                    .onErrorReturn { Optional.empty() }
+                                    .onErrorReturn {
+                                        // Profile API can be restricted and doesn't have to return result.
+                                        // In this case allow inviting valid user ids.
+                                        User(
+                                                userId = search,
+                                                displayName = search,
```
[the remark this answers is not in the document]
Comment (onurays) [reply vector/src/main/java/im/vector/app/features/userdirectory/UserListViewModel.kt]: Done (web is not doing so though).

**Pull request merged**